src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -259,22 +259,18 @@ void CouplingSchemeConfiguration::xmlTagCallback(
     bool          initialize          = tag.getBooleanAttributeValue(ATTR_INITIALIZE);
     mesh::PtrData exchangeData;
     mesh::PtrMesh exchangeMesh;
-    for (const mesh::PtrMesh &mesh : _meshConfig->meshes()) {
-      if (mesh->getName() == nameMesh) {
-        for (const mesh::PtrData &data : mesh->data()) {
-          if (data->getName() == nameData) {
-            exchangeData = data;
-            exchangeMesh = mesh;
-            break;
-          }
-        }
-      }
-    }
-    PRECICE_CHECK(exchangeData.get(),
+
+    PRECICE_CHECK(_meshConfig->hasMeshName(nameMesh) && _meshConfig->getMesh(nameMesh)->hasDataName(nameData),
                   "Mesh \"{}\" with data \"{}\" not defined. "
                   "Please check the <exchange data=\"{}\" mesh=\"{}\" from=\"{}\" to=\"{}\" /> "
                   "tag in the <coupling-scheme:... /> of your precice-config.xml.",
                   nameMesh, nameData, nameData, nameMesh, nameParticipantFrom, nameParticipantTo);
+
+    const mesh::PtrMesh &mesh = _meshConfig->getMesh(nameMesh);
+    const mesh::PtrData &data = mesh->data(nameData);
```

Comment:
You could `PRECICE_ASSERT(mesh && data);` to really make sure nothing weird happens here.

Reply:
Done 4810dbbb4dda719a0fb676ac7e7c0efbf7c6ca76.